File: web/CookShare/src/types/recipe.ts
export type Difficulty = 'Easy' | 'Medium' | 'Hard';

export interface Comment {
  id: number;
  author: string;
  text: string;
  date: string;
}

export interface Recipe {
  id: number;
  title: string;
  description: string;
  tags: string[];
  rating: number;
  reviewCount: number;
  prepTime: string;
  cookTime: string;
  difficulty: Difficulty;
  author: string;
  servings: number;
  imageUrl: string;
  ingredients: string[];
  instructions: string[];
  postedDate: string;
  comments: Comment[];
}

File: mobile/app/src/main/java/com/salgado/cookshare/model/Recipe.kt
package com.salgado.cookshare.model

data class Recipe(
    val id: String,
    val title: String,
    val description: String,
    val tags: List<String>,
    val rating: Double,
    val reviewCount: Int,
    val prepTime: String,
    val cookTime: String,
    val difficulty: String,
    val author: String,
    val servings: Int,
    val imageUrl: String,
    val ingredients: List<String>,
    val instructions: List<String>,
    val category: String = ""
)